### DISCUSSION TITLE
MQTT Sensor in HA has "---" state

### DISCUSSION THREAD

**Discussion by proasnet** [fredlcore/BSB-LAN#614] [Q&A] [answered]
Please, help me with MQTT sensor in HA. Some values from BSBLAN has after off status not zero values.
"---", not "0", as burner modulation, pump speed...

![obrazek](https://github.com/fredlcore/BSB-LAN/assets/76579362/208c31d4-f82b-4769-ae07-42eeb0029e8e)

```
  - state_topic: "BSB-LAN/8308"
    name: BSB-LAN Boiler pump speed
    unique_id: bsb_lan_boiler_pump_speed
    availability_topic: "BSB-LAN/status"
    unit_of_measurement: "%"
    device_class: power_factor
```

Is possible define in yaml mqtt sensor, that "---" is zero? In HA, after message is "---", is displayed last value of sensor, not actual.
Thank you.

**Comment by fredlcore** [comment, marked as answer]
Zero is not the same as `---`. For a sensor, this means that the sensor is either not available or deactivated, so interpreting this as 0 could have severe consequences. Imagine you have a temperature sensor which is deactivated and the system would act as if it was at 0 degrees...

**Reply by proasnet** [reply]
I understand the interpretation that it is not the same thing to have --- and 0, on the driving side.
But in my case it is a sensor in HA, which does not follow any serious management. So 0 instead of --- won't hurt anything.
This is not an adjustment on the BSBLAN side, but an evaluation --- on the HA side.

**Comment by fredlcore** [comment]
Ah, ok, I'm sorry that I can't help you then as I'm not a HA user/expert...

**Reply by proasnet** [reply]
Can you please open a topic for me? Maybe someone can advise me. Thanks
